api/kroxylicious-filter-api/src/main/java/io/kroxylicious/proxy/filter/FilterInvokers.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.message.ResponseHeaderData;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ApiMessage;
+
+/**
+ * Factory for FilterInvokers. The intention is to keep the Invoker implementations
+ * as private as we can, so that invocation is a framework concern.
+ */
+public class FilterInvokers {
```

Comment:
I've kept this responsibility in the `kroxylicious-filter-api` module for now because the multi-tenant unit test wants a convenient way to invoke the Filter for lots of different ApiMessage types. I think this is an okay first step but maybe we could in future move this into `kroxylicious` and access it through some kind of testing harness.

Reply:
I'm thinking of something like the multitenant module having a test dependency on a `kroxylicious-test` module that makes it convenient to test your Filter with some of the real Kroxy invocation code

Reply:
Sounds reasonable. I agree that it could wait to a later step.